action_plugins/load_profile/__init__.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8; -*-
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from __future__ import annotations
+
+import os
+import logging
+from typing import List, TYPE_CHECKING
+from xml.etree import ElementTree
+
+from PySide6 import QtCore
+from PySide6.QtCore import Property, Signal
+
+from gremlin import event_handler, util
+from gremlin.base_classes import AbstractActionData, AbstractFunctor, \
+    Value
+
+from gremlin.profile import Library
+from gremlin.types import ActionProperty, InputType, PropertyType
+
+from gremlin.ui.action_model import SequenceIndex, ActionModel
+from gremlin.ui import backend
+from gremlin.error import GremlinError
+
+if TYPE_CHECKING:
+    from gremlin.ui.profile import InputItemBindingModel
+
+
+class LoadProfileFunctor(AbstractFunctor):
+
+    """Executes a load profile action callback."""
+
+    def __init__(self, action: LoadProfileData):
+        super().__init__(action)
+
+    def __call__(
+        self,
+        event: event_handler.Event,
+        value: Value
+    ) -> None:
+
+        if not self._should_execute(value):
+            return
+
+        logging.getLogger("system").debug(
+            f"Loading profile ...{self.data.profile_filename}"
+        )
+
+        be = backend.Backend()
```

Comment:
Maybe worth checking the file is still valid here or in the `_from_xml()` (likely better since I hope people don't move a profile when they actively use Gremlin) call to prevent attempting to load an inaccessible one. I think you should be fine with regards to profile errors as that error chain leads to a UI error dialogue. Starting can throw exceptions and they might be uncaught until termination possibly.

Reply:
Added validator to _from_xml()